sql/migrate/dir.go

```diff
@@ -555,3 +556,73 @@ func (m *memFile) Mode() fs.FileMode          { return 0 }
 func (m *memFile) ModTime() time.Time         { return time.Time{} }
 func (m *memFile) IsDir() bool                { return false }
 func (m *memFile) Sys() interface{}           { return nil }
+
+// ArchiveDir returns a tar archive of the given directory.
+func ArchiveDir(dir Dir) ([]byte, error) {
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	defer tw.Close()
+
+	sumF, err := dir.Open(HashFileName)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
```

Comment:
```suggestion
	if err != nil {
```
We can keep it simple and fail here.

Reply:
I want to skip errnotexist so we can archive directories without sumfiles, shouldn't we support that ?

Reply:
kk